src/plugins/intel_cpu/src/nodes/inverse.cpp

```diff
@@ -0,0 +1,257 @@
+#include <openvino/op/constant.hpp>
+#include <openvino/op/inverse.hpp>
+
+#include "utils/bfloat16.hpp"
+
+// Parallel LU decomposition algorithm with partial pivoting
+// Based on the lectures by Prof. Dr. Thomas Huckle, Parallel Numerics
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+Inverse::Inverse(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op, EMPTY_PORT_MASK)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        THROW_CPU_NODE_ERR(errorMessage);
+    }
+
+    auto inverse_op = as_type_ptr<op::v14::Inverse>(op);
+    m_adjoint = inverse_op->get_adjoint();
+
+    constant = ConstantType::StrictNoConst;
+
+    m_const_input = is_type<op::v0::Constant>(op->get_input_node_ptr(INPUT_PORT));
+}
+
+bool Inverse::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (op->get_type_info() != op::v14::Inverse::get_type_info_static()) {
+            errorMessage = "Only Inverse operation from the opset14 is supported by the CPU plugin.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void Inverse::getSupportedDescriptors() {
+    if (getParentEdges().size() != 1) {
+        THROW_CPU_NODE_ERR("has incorrect number of input edges.");
+    }
+    if (getChildEdges().empty()) {
+        THROW_CPU_NODE_ERR("has incorrect number of output edges.");
+    }
+}
+
+void Inverse::initSupportedPrimitiveDescriptors() {
+    m_input_precision = getOriginalInputPrecisionAtPort(INPUT_PORT);
+    if (!one_of(m_input_precision, ov::element::f32, ov::element::f16, ov::element::bf16)) {
+        m_input_precision = ov::element::f32;
+    }
+
+    addSupportedPrimDesc({{LayoutType::ncsp, m_input_precision, m_const_input}},
+                         {{LayoutType::ncsp, m_input_precision}},
+                         ref_any);
+}
+
+bool Inverse::needShapeInfer() const {
+    return !m_const_input;
+}
+
+bool Inverse::needPrepareParams() const {
+    return true;
+}
+
+void Inverse::prepareParams() {
+    const auto& input_shape = getParentEdgeAt(INPUT_PORT)->getMemory().getStaticDims();
+
+    if (input_shape.size() < 2) {
+        THROW_CPU_NODE_ERR("has incompatible 'data' shape ",
+                           PartialShape(input_shape),
+                           ". Only tensors of rank at least 2 are allowed.");
+    }
+
+    m_side = input_shape.back();
+    m_side_squared = m_side * m_side;
+    m_batches_count = 1;
+
+    for (size_t i = 0; i < input_shape.size() - 2; ++i) {
+        m_batches_count = m_batches_count * input_shape[i];
+    }
+}
+
+bool Inverse::isExecutable() const {
+    return !isInputTensorAtPortEmpty(INPUT_PORT);
+}
+
+bool Inverse::created() const {
+    return getType() == Type::Inverse;
+}
+
+void Inverse::execute(dnnl::stream strm) {
+    switch (m_input_precision) {
```

Comment:
Please use OV_SWITCH and OV_CASE like [here](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/cum_sum.cpp#L98)

Reply:
Used!